R/apa_print_htest.R

```diff
@@ -79,76 +79,84 @@ apa_print.htest <- function(
 
   ellipsis <- list(...)
 
-  if(is.null(stat_name) & !is.null(names(x$statistic))) {
-    stat_name <- names(x$statistic)
-    stat_name <- convert_stat_name(stat_name)
+
+  # Arrange table, i.e. coerce htest to a proper data frame ----
+
+  if(length(x$estimate) == 2L) {
+    x$estimate <- unname(diff(rev(x$estimate)))
+    names(x$estimate) <- names(x$null.value)
   }
-  stat <- printnum(x$statistic)
-
-  if(!is.null(x$sample.size)) n <- x$sample.size
-
-  if(!is.null(x$parameter)) {
-    # Statistic and degrees of freedom
-    parameter_names <- tolower(names(x$parameter))
-    if(length(parameter_names) == 1 && parameter_names == "df") {
-      dfdigits <- (x$parameter %%1 != 0) * 2
-      if(stat_name == "\\chi^2") {
-        if(is.null(x$sample.size) & is.null(n)) stop("Please provide the sample size to report.") # Demand sample size information if it's a Chi^2 test
-        stat_name <- paste0(stat_name, "(", printnum(x$parameter[grep("df", parameter_names)], digits = dfdigits), ", n = ", n, ")")
-      } else {
-        stat_name <- paste0(stat_name, "(", printnum(x$parameter[grep("df", parameter_names)], digits = dfdigits), ")")
-      }
-    } else if(length(parameter_names) == 2 && identical(parameter_names, c("num df", "denom df"))) {
-      dfdigits <- (x$parameter %%1 != 0) * 2
-      stat_name <- paste0(stat_name, "(", printnum(x$parameter[grep("num df", parameter_names)], digits = dfdigits[1]), ", ", printnum(x$parameter[grep("denom df", parameter_names)], digits = dfdigits[2]), ")")
-    }
+  if(length(x$estimate) > 2L) x$estimate <- NULL
+
+  if(is.null(ci)) {
+    conf_int <- list(x$conf.int)
+  } else {
+    conf_int <- list(ci)
   }
 
-  # p-value
-  p <- printp(x$p.value)
 
-  apa_res <- apa_print_container()
-  apa_res$statistic <- paste0("$", stat_name, " = ", stat, "$, $p ", add_equals(p), "$")
-  if(in_paren) apa_res$statistic <- in_paren(apa_res$statistic)
 
-  # Estimate
-  if(is.null(est_name) & !is.null(names(x$estimate))) {
-    est_name <- convert_stat_name(names(x$estimate))
-  }
+  if(is.null(x$parameter)) x$parameter <- NULL
 
-  if(is.null(x$estimate)) {
-    est <- NULL
-  } else {
-    if(is.null(est_name)) {
-      warning("Cannot determine name of estimate supplied in ", deparse(substitute(x)), " of class 'htest'. Estimate is omitted from output string. Please set parameter 'est_name' to obtain an estimate.")
-      est <- NULL
-    } else if(!is.null(names(x$estimate)) && est_name %in% c("\\Delta M", "\\Delta p")) {
-      est <- do.call(function(...) printnum(diff(rev(x$estimate)), ...), ellipsis)
-    } else if(length(x$estimate) == 1) {
-      if(est_name %in% c("r", "r_{\\mathrm{s}}", "\\uptau") & is.null(ellipsis$gt1)) ellipsis$gt1 <- FALSE
-      est <- do.call(function(...) printnum(x$estimate, ...), ellipsis)
+  x$conf.int    <- NULL
+  x$null.value  <- NULL
+  x$alternative <- NULL
+  x$method      <- NULL
+  x$data.name   <- NULL
+
+  x_list <- lapply(x, FUN = function(x) {
```

Comment:
I turn each element into a single-row matrix because this ensures that `as.data.frame()` will always create a `data.frame` with only one row; I thought there might be some `htest` objects out there that have elements with length > 1 we don't know of, yet.